third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -887,6 +895,78 @@ struct AtomicRMWOpConversion
         continue;
       }
 
+      // Let LLVM handle compare+swap loop; branch-based pred should be fine
+      if (doLLVMFallback) {
```

Comment:
Since this function is already quite long and complicated, would it make sense to factor the llvm fallback into a helper, for clarity?  Of course maybe not if you'd have to pass a ton of state, but worth checking.